### PR TITLE
update katex to version 0.8.3

### PR DESCRIPTION
**Pull request by pitamar**
@RocketChat/core

### CODE REVIEW

**Comment by pitamar** [comment]
@rodrigok 
Hi, it's been a while since I opened the PR. Is there something missing?
Thanks

**Comment by geekgonecrazy** [comment]
@pitamar Thanks for the PR.  Can you summarize a bit more whats in the update?  

Katex is one of those things most of us aren't super familiar with making use of.  I think in the past we've actually had to downgrade a few packages of this nature because of performance issues.

**Comment by pitamar** [comment]
Hi @geekgonecrazy 

The currently used version of Katex in RocketChat is 0.7.1.
There has been many updates to the library since though.
I concentrated them here:

> # v0.8.3 (2017-08-27)
> ## New features:
> * Add latin-1 letters as math symbols (#796)
> * Support CJK full-width punctuation + Unicode dots (#814)
> * Support for ' ` ^ ~ = \u . " \r \H \v text-mode accents (#802)
> * Added support for \not (#140)
> 
> ## Bug fixes:
> * Use inline SVG for stretchy elements (#807)
> * Improve \sqrt #810
> 
> ## Other changes:
> * Modernize font creation (#624)
> * Pre-commit hook to call make lint (#815)
> * Fix link in CONTRIBUTING.md.
> * Reduce SVG (#819)
> * switch pre-commit hook to use pre-commit npm package (#823)
> 
> # v0.8.2 (2017-08-17)
> ## New features:
> * Accept all existing Greek letters using unicode characters in math mode (#410)
> 
> ## Bug fixes:
> * Fix MathML output for ' and large operators with limits (#788)
> * Update package.json to point 'main' at dist/katex.js (#791)
> * Fix color support for stretchy, strikethrough, and fbox (#792)
> 
> ## Other changes:
> * Detect attachEvent() support correctly, fixes #771 (#772)
> 
> # v0.8.1 (2017-08-11)
> The v0.8.0 release did not include the SVG images which are necessary for rendering wide and stretchy accents as well as \overbrace and \underbrace. This release corrects that.
> 
> Note that if you're including copies of KaTeX in your web application, you should now include the dist/images directory in addition to the usual the dist/fonts directory.
> 
> Many thanks to @spicyj for fixing the issue so quickly.
> 
> # v0.8.0 (2017-08-11)
> ## Breaking changes:
> * Implicit \color, explicitly grouped \textcolor #619
> 
> ## New features:
> * fix #509: added some international operators #647
> * Old font command support: \rm, \sf, \tt, \bf, \it #675
> * Builtin macros, macro arguments, \overset and \underset #605
> * Add \iff, \implies, \impliedby support #697
> * Support <, >, | and many \text... commands in text mode #684
> * Implement $...$ via styling node #637
> * Support Windows high-contrast mode. Fix #716 #724
> * Add \jot lineskip to aligned environment, switch contents to displaystyle, and add gathered #725
> * Fix high contrast mode better, thanks to @GeeLaw #733
> * Support stretchy wide elements. #670
> * Implement \coloneqq, \colonequals, etc. based on mathtools and colonequals #727
> * Add configurable error callback #658
> * Support absolute TeX units #732
> 
> ## Other changes:
> * Revert "Remove trailing commas for IE 9 compatibility" #622
> * Fix all AMS mathord symbols #618
> * Fix x'^2 #636
> * Use utils.deflt for Settings #649
> * Fix font typo math -> main #678
> * Fix spaces before \middle #689
> * Refactor and comment space splicing code #699
> * Fix #711 issue with multiple superscripts. #718
> * Fix interaction between styles and sizes. #719
> * Vertically center single-character \mathop. #745
> * Correct handling of unbraced kerns followed by spaces. #751
> * Associate font metrics with Options, not Style. #743
> * Upgrade the source to use ES6 syntax including classes, import and static properties #679
> * Set maxFontSize on rules. #744
> * Correct computation of TeX sizes. #755
> * Use \displaystyle within \over/\underbrace #765
> * Shrinkwrap vlists in table-like CSS. #768
> * Solve Safari rendering issues with font-size overrides. #780
> * Improve rule coding, including for \sqrt. #776
>
> ## Infra changes:
> * Update dependencies to more recent version #612
> * Use morgan to log requests in the dev server #625
> * Improve release script and bower support #615
> * Show compressed sizes correctly #639
> * Set up a diff attribute and textconv instructions for font files #635
> * Fix release script after experience from 0.7.1 release #642
> * Update texcmp to ubuntu 17.04 and avoid mounted host directory #722
> * Screenshotter: Add --diff and --attempts options. #736
> * Switch from jasmine to jest #747
> * Ensure screenshotter runs all tests. #769
> 
> ## Misc:
> * Describe using auto renderer in html head #623
> * README.md - added integrity #590
> * README.md - added integrity #591
> * Switch speed test in README #640
> * Describe site-provided macros in README #644
> * Fixed missing verb in README #668
> * Rephrase displayMode description to be clearer #648
> * Add documentation for how to run auto-render example #740
> * Mention texcmp in CONTRIBUTING.md #753

As you can can see most of the changes are bug fixes or addition of new features (like new symbols).
All in all, besides many improvements, it doesn't seem to be a dramatic change.
I didn't feel any slowdown using the new version of Katex. Are there any specific tests you'd like me to perform?

Thanks =)
Itamar

**Comment by pitamar** [comment]
Is there anything I can do to help?
Do you prefer not to upgrade the katex version? is there a reason to it?

@RocketChat/core

**Comment by engelgabriel** [comment]
Maybe we should use the version v0.9.0

https://github.com/Khan/KaTeX/releases

**Comment by pitamar** [comment]
ye, that version was the most recent at the time. if you want i update the pull request for 0.9.0

**Comment by engelgabriel** [comment]
@pitamar Please do so we can merge it.

**Comment by pitamar** [comment]
@engelgabriel Ok, I pushed a new commit for version 0.9.0. seems to work fine to me
